src/main/java/use_cases/course_features/course_enrolment_use_case/CourseEnrolmentInteractor.java

```diff
@@ -84,39 +85,44 @@ public CourseEnrolmentResponseModel enrol(CourseEnrolmentRequestModel requestMod
             courseTaskId.add(taskTitleToId);
```

Comment:
above on line 46/47, I think it should be `if (!courseEnrolmentDsGateway.existsByCourseID(requestModel.getCourseID())) {` (i.e. if course does NOT exist)

Reply:
Fixed! Thanks for catching it